cmd/az-log/cmd/common.go

```diff
@@ -0,0 +1,241 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+
+	homedir "github.com/mitchellh/go-homedir"
+	"github.com/spf13/cobra"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	consts "sigs.k8s.io/azuredisk-csi-driver/pkg/azureconstants"
+)
+
+const (
+	AzureDiskContainer = "azuredisk"
+	RFC3339Format      = `^\d{4}-(\d{2})-(\d{2})T(\d{2}:\d{2}:\d{2}(.\d+)?)`
+	KlogTimeFormat     = `(\d{4}) (\d{2}:\d{2}:\d{2}(.\d+)?)`
+)
+
+func GetFlags(cmd *cobra.Command) ([]string, []string, []string, string, string, bool, bool) {
+	volumes, _ := cmd.Flags().GetStringSlice("volume")
+	nodes, _ := cmd.Flags().GetStringSlice("node")
+	requestIds, _ := cmd.Flags().GetStringSlice("request-id")
+	since, _ := cmd.Flags().GetString("since")
+	sinceTime, _ := cmd.Flags().GetString("since-time")
+	isFollow, _ := cmd.Flags().GetBool("follow")
+	isPrevious, _ := cmd.Flags().GetBool("previous")
+
+	if since != "" && sinceTime != "" {
+		fmt.Println("error: only one of --since/--since-time may be specified")
+		os.Exit(0)
+	}
+
+	return volumes, nodes, requestIds, since, sinceTime, isFollow, isPrevious
+}
+
+func getConfig() *rest.Config {
+	var kubeconfig string
+	if home, _ := homedir.Dir(); home != "" {
+		kubeconfig = filepath.Join(home, ".kube", "config")
+	}
+
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	return config
+}
+
+func getKubernetesClientset(config *rest.Config) *kubernetes.Clientset {
+	clientsetK8s, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err.Error())
+	}
+	return clientsetK8s
+}
+
+func GetLogsByAzDriverPod(clientsetK8s kubernetes.Interface, podName string, container string, volumes []string,
```

Comment:
Should we move this to pod.go instead to keep it consistent? Since we have the GetLogsByFile under the file.go?